lib/omnibus/cli.rb

```diff
@@ -61,12 +61,32 @@ def execute!
     #
     #   $ omnibus build chefdk
     #
+    method_option :output_manifest,
+      desc: "Create version-manifest.json in current directory at the end of the build",
+      type: :boolean,
+      default: false
+    method_option :use_manifest,
+      desc: "Use the given manifest when downloading software sources.",
+      type: :string,
+      default: nil
     desc 'build PROJECT', 'Build the given Omnibus project'
     def build(name)
-      project = Project.load(name)
+      manifest = if @options[:use_manifest]
+                   Omnibus::Manifest.from_file(@options[:use_manifest])
+                 else
+                   nil
+                 end
 
+      project = Project.load(name, manifest)
       say("Building #{project.name} #{project.build_version}...")
-      project.build_me
+      project.download
+      project.build
+
+      if @options[:output_manifest]
+        File.open('version-manifest.json', 'w') do |f|
+          f.write(JSON.pretty_generate(project.built_manifest.to_hash))
+        end
+      end
```

Comment:
It is a bit round-about, but built_manifest shouldn't recalculate anything.  Project#built_manifest calls Software.manifest_entry

https://github.com/chef/omnibus/pull/430/files#diff-9d12b132bef62277d2011f5c7bf2368cR970

Software#manifest_entry is memoized at the time of creation:

https://github.com/chef/omnibus/pull/430/files#diff-2753487ba152ae43f78b4da0c96b8e44R103

So it will always return either the manifest entry it was passed on instantiation or the one it generated itself during the build.